From 9be53958658b6871a5af0c11c785cd98c5e49ed0 Mon Sep 17 00:00:00 2001
From: Marius Brehler <marius.brehler@amd.com>
Date: Mon, 28 Apr 2025 14:46:44 +0000
Subject: [PATCH 1/2] Set `GTEST_BOTH_LIBRARIES` appropriately

If `find_package()` succeeds to find GTest and `INSTALL_DEPENDENCIES`
is set to OFF, `GTEST_BOTH_LIBRARIES` is not set and thus
`rccl-UnitTests` fails with trying to link unkown symbols.
---
 cmake/Dependencies.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index d7095b38..0d54c20d 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -68,6 +68,8 @@ if(NOT GTest_FOUND AND BUILD_TESTS OR INSTALL_DEPENDENCIES)
         message(FATAL_ERROR "Cannot find gtest library installation path.")
     find_package(GTest REQUIRED CONFIG PATHS ${GTEST_ROOT})
     endif()
+elseif(GTest_FOUND AND BUILD_TESTS)
+  set(GTEST_BOTH_LIBRARIES "GTest::gtest;GTest::gtest_main")
 endif()
 
 # Find or download/install rocm-cmake project
-- 
2.43.0

